memory/table.go

```diff
@@ -486,41 +504,87 @@ func (t *Table) insertValueInRows(ctx *sql.Context, idx int, colDefault *sql.Col
 }
 
 func (t *Table) DropColumn(ctx *sql.Context, columnName string) error {
+	droppedCol := t.dropColumnFromSchema(ctx, columnName)
+	for k, p := range t.partitions {
+		newP := make([]sql.Row, len(p))
+		for i, row := range p {
+			var newRow sql.Row
+			newRow = append(newRow, row[:droppedCol]...)
+			newRow = append(newRow, row[droppedCol+1:]...)
+			newP[i] = newRow
+		}
+		t.partitions[k] = newP
+	}
+	return nil
+}
+
+// dropColumnFromSchema drops the given column name from the schema and returns its old index.
+func (t *Table) dropColumnFromSchema(ctx *sql.Context, columnName string) int {
 	newSch := make(sql.Schema, len(t.schema)-1)
 	var i int
+	droppedCol := -1
 	for _, col := range t.schema {
 		if col.Name != columnName {
 			newSch[i] = col
 			i++
+		} else {
+			droppedCol = i
 		}
 	}
 	t.schema = newSch
-	return nil
+	return droppedCol
 }
 
 func (t *Table) ModifyColumn(ctx *sql.Context, columnName string, column *sql.Column, order *sql.ColumnOrder) error {
```

Comment:
These things are honestly the worst, thanks for fixing them

Reply:
There was a lot of pain involved in this file lol